FormidableLabs/react-native-app-auth
# Add claims parameter to authorize

## Description

Currently, the `setClaims` method of App Auth is not exposed. By adding a new key `claims` to `AuthConfig`, similar to `scopes`, when it is set, we can call `setClaims` on the `AuthorizationRequest.Builder` to add the requested claims to the authorization request.

## Steps to verify

Create a `AuthConfiguration` that requests some claims about the user:

```typescript
const config: AuthConfiguration = {
  issuer: 'http://example',
  clientId: '....',
  redirectUrl: 'localhost:8080',
  scopes: ['openid', 'offline_access'],
  claims: { "id_token": { "email": null } }
}
```
Then, use `authorize` to obtain an idToken, and verify that the requested claims are included in the token.

## Draft

This PR implements the `claims` key in the `AuthConfiguration` on the Android side. However, I do not own iOS devices and can therefore not implement this for iOS. I am opening this PR in the hopes that someone else completes it for the iOS side.
